nncf/quantization/algorithms/channel_alignment/backend.py

```diff
@@ -0,0 +1,150 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Optional, Tuple, TypeVar
+
+import numpy as np
+
+from nncf.common.graph import NNCFGraph
+from nncf.common.graph import NNCFNode
+from nncf.common.graph.layer_attributes import ConvolutionLayerAttributes
+from nncf.common.graph.transformations.commands import TargetPoint
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.tensor_statistics.collectors import TensorStatisticCollectorBase
+from nncf.common.utils.registry import Registry
+
+TModel = TypeVar("TModel")
+OutputType = TypeVar("OutputType")
+ALGO_BACKENDS = Registry("algo_backends")
+
+
+@dataclass
+class DimsDescriptor:
+    conv_weight_out_channels_dim: int
+    conv_weight_in_channels_dim: int
+    bias_channels_dim: int
+
+
+class ChannelAlignmentAlgoBackend:
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> TargetPoint:
+        """
+        Returns backend-specific target point.
+
+        :param target_type: Type of the location that should be modified.
+        :param target_node_name: Name of the located node.
+        :param port_id: id of the port for the statistics disctribution.
+        :return: Backend-specific TargetPoint.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_bias_value(node: NNCFNode, model: TModel, nncf_graph: NNCFGraph) -> np.ndarray:
+        """
+        Returns bias value in the NumPy format of provided node.
+
+        :param node: Node of NNCFGraph with bias value.
+        :param model: Backend-specific model for the initializer finding.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weight_value(node: NNCFNode, model: TModel, port_id: int) -> np.ndarray:
+        """
+        Returns bias value in the NumPy format of provided node.
+
+        :param node: Node of NNCFGraph with bias value.
+        :param model: Backend-specific model for the initializer finding.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to activation input and output edges for
+        the node.
+        Supports only nodes that could have bias value.
+
+        :param node: Node of NNCFGraph with bias value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        pass
```

Comment:
Add docstrings for all methods in  ChannelAlignmentAlgoBackend

Reply:
Done, please check